src/v/cluster/topic_table.cc

```diff
@@ -326,9 +337,11 @@ topic_table::apply(finish_moving_partition_replicas_cmd cmd, model::offset o) {
     auto found = _topics_hierarchy.find(model::topic_namespace_view(cmd.key));
     if (found != _topics_hierarchy.end()) {
         for (const auto& cs : found->second) {
-            bool erased = _updates_in_progress.erase(
-                            model::ntp(cs.ns, cs.tp, cmd.key.tp.partition))
-                          > 0;
+            auto ntp = model::ntp(cs.ns, cs.tp, cmd.key.tp.partition);
+            if (_updates_in_progress.contains(ntp)) {
+                _probe.handle_update_removal(ntp);
```

Comment:
would it be possible to wrap this in some kind of an RAII object to make the updates automatic ? Maybe we can make this part of the `_updates_in_progress` map value ?